### PR TITLE
pmd: bump to 6.14.0

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

**Comment by LegacyKing** [comment]
@grimreaper broken...

File: code/src/java/pcgen/core/AbilityUtilities.java [hunk 1]
```diff
@@ -113,7 +113,6 @@ public static String removeChoicesFromName(String name)
 	 * @throws GroupingMismatchException If there are mismatched brackets
 	 */
 	public static String getUndecoratedName(final String name, final Collection<String> specifics)
-		throws GroupingMismatchException
```
Comment (thpr) [inline code/src/java/pcgen/core/AbilityUtilities.java]: Now inconsistent with the docs of the method 3 lines up...

File: code/src/java/pcgen/core/chooser/InfoWrapper.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
  */
 public class InfoWrapper implements InfoFacade, SortKeyAware
 {
-	private static final NumberFormat SORTABLE_NUMBER_FORMAT = new DecimalFormat("0000000000.00000");
+	private final NumberFormat SORTABLE_NUMBER_FORMAT = new DecimalFormat("0000000000.00000");
```
Comment (thpr) [inline code/src/java/pcgen/core/chooser/InfoWrapper.java]: Why remove static?

File: code/src/java/pcgen/gui2/converter/panel/RunConvertPanel.java [hunk 3]
```diff
@@ -131,93 +131,88 @@ public boolean performAnalysis(final CDOMObject pc)
 		}
 		sortCampaignsByRank(totalCampaigns);
 
-		new Thread(new Runnable()
-		{
-			@Override
-			public void run()
+		new Thread(() -> {
```
Comment (thpr) [inline code/src/java/pcgen/gui2/converter/panel/RunConvertPanel.java]: IMHO taking a huge anonymous thing and making another huge anonymous thing is kind of meh.  Something that is almost 100 lines long probably belongs in a named class? Thoughts?

File: code/src/java/pcgen/gui2/facade/Gui2InfoFactory.java [hunk 4]
```diff
@@ -125,15 +126,13 @@
...
- *
- * 
  */
 public class Gui2InfoFactory implements InfoFactory
 {
 	/** A default return value for an invalid request. */
 	private static final String EMPTY_STRING = ""; //$NON-NLS-1$
-	private static final NumberFormat ADJ_FMT = new DecimalFormat("+0;-0"); //$NON-NLS-1$
-	private static final NumberFormat COST_FMT = new DecimalFormat("#,##0.##"); //$NON-NLS-1$
+	private final Format ADJ_FMT = new DecimalFormat("+0;-0"); //$NON-NLS-1$
```
Comment (thpr) [inline code/src/java/pcgen/gui2/facade/Gui2InfoFactory.java]: I still don't understand the value of this change.  If it is thrown around as merely a Format, we lose methods on the interface and it decreases the amount of semantic information available to the developer.  I see these as negative value changes.  YMMV.

File: code/src/java/plugin/initiative/gui/OpposedCheckDialog.java [hunk 5]
```diff
@@ -141,7 +141,6 @@
 	 * @throws HeadlessException if running without a gui
 	 */
 	OpposedCheckDialog(Frame owner, List<InitHolder> rollingGroup, List<InitHolder> availableGroup)
-		throws HeadlessException
```
Comment (thpr) [inline code/src/java/plugin/initiative/gui/OpposedCheckDialog.java]: Now inconsistent with JavaDoc above

File: code/src/java/plugin/initiative/gui/DamageDialog.java [hunk 6]
```diff
@@ -75,7 +75,7 @@ public boolean getSubdual()
 	 * Check damage field
 	 * @throws NumberFormatException
 	 */
-	private void checkDamageField() throws NumberFormatException
+	private void checkDamageField()
```
Comment (thpr) [inline code/src/java/plugin/initiative/gui/DamageDialog.java]: Inconsistent with JavaDocs

File: code/src/java/pcgen/gui2/util/GridBoxLayout.java [hunk 7]
```diff
@@ -50,7 +50,7 @@ public GridBoxLayout(int rows, int cols)
 	 * @param vgap
 	 * @throws IllegalArgumentException  if the value of both rows and cols is set to zero
 	 */
-	public GridBoxLayout(int rows, int cols, int hgap, int vgap) throws IllegalArgumentException
+	public GridBoxLayout(int rows, int cols, int hgap, int vgap)
```
Comment (thpr) [inline code/src/java/pcgen/gui2/util/GridBoxLayout.java]: Inconsistent with JavaDocs

File: code/src/java/pcgen/gui2/facade/CharacterFacadeImpl.java [hunk 8]
```diff
@@ -2531,7 +2531,7 @@ public UIDelegate getUIDelegate()
 	 * @throws NullPointerException 
 	 * @throws IOException If the write fails
 	 */
-	public void save() throws NullPointerException, IOException
+	public void save() throws IOException
```
Comment (thpr) [inline code/src/java/pcgen/gui2/facade/CharacterFacadeImpl.java]: Inconsistent with JavaDocs

File: code/src/java/pcgen/core/AbilityUtilities.java (hunk 1 above)
Comment (grimreaper) [reply code/src/java/pcgen/core/AbilityUtilities.java]: It is not inconsistent. Instead, the documentation serves the role of documentation.
See https://pmd.github.io/latest/pmd_rules_java_design.html#avoiduncheckedexceptionsinsignatures for the reasoning.

**Comment by grimreaper** [comment]
recreating

**Pull request closed without merging**